x/did/client/rest/tx.go

```diff
@@ -12,38 +14,37 @@ import (
 	"github.com/ixofoundation/ixo-blockchain/x/did/internal/keeper"
 	"github.com/ixofoundation/ixo-blockchain/x/did/internal/types"
 	"github.com/ixofoundation/ixo-blockchain/x/ixo"
+
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
 )
 
 func registerTxRoutes(cliCtx context.CLIContext, r *mux.Router) {
 	r.HandleFunc("/did", createDidRequestHandler(cliCtx)).Methods("POST")
 	r.HandleFunc("/credential", addCredentialRequestHandler(cliCtx)).Methods("POST")
 }
 
+type MsgAddDid struct {
+	BaseReq rest.BaseReq `json:"base_req" yaml:"base_req"`
+	Did     exported.Did `json:"did" yaml:"did"`
+	PubKey  string       `json:"pubKey" yaml:"pubKey"`
+}
+
 func createDidRequestHandler(cliCtx context.CLIContext) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 
-		w.Header().Set("Content-Type", "application/json")
-		didDocParam := r.URL.Query().Get("didDoc")
-		mode := r.URL.Query().Get("mode")
-		cliCtx = cliCtx.WithBroadcastMode(mode)
+		var req MsgAddDid
 
-		ixoDid, err := types.UnmarshalIxoDid(didDocParam)
-		if err != nil {
-			w.WriteHeader(http.StatusBadRequest)
-			_, _ = w.Write([]byte(err.Error()))
+		if !rest.ReadRESTReq(w, r, cliCtx.Codec, &req) {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, "failed to parse request")
```

Comment:
The `ReadRESTReq` function writes an error response itself. So we can remove this line.

Reply:
removed